tests/functional/refine.t

```diff
@@ -106,4 +106,50 @@ This approach does not make sense and should not work without an alignment FASTA
   *ERROR: alignment is required* (glob)
   [1]
 
+Ensure --year-bounds behaves as expected.
+
+Create a copy of tests/functional/refine/metadata.tsv, adding partial ambiguity on the century-level (20XX) for the first strain PAN/CDC_259359_V1_V3/2015.
+
+  $ cat >"$TMP/metadata.tsv" <<~~
+  > strain	date
+  > PAN/CDC_259359_V1_V3/2015	20XX-XX-XX
+  > COL/FLR_00024/2015	2015-12-XX
+  > PRVABC59	2015-12-XX
+  > COL/FLR_00008/2015	2015-12-XX
+  > Colombia/2016/ZC204Se	2016-01-06
+  > ZKC2/2016	2016-02-16
+  > VEN/UF_1/2016	2016-03-25
+  > DOM/2016/BB_0059	2016-04-04
+  > BRA/2016/FC_6706	2016-04-08
+  > DOM/2016/BB_0183	2016-04-18
+  > EcEs062_16	2016-04-XX
+  > HND/2016/HU_ME59	2016-05-13
+  > ~~
+
+Limit ambiguous dates to be within (2000, 2020).
+
+  $ ${AUGUR} refine \
+  >  --tree "refine/tree_raw.nwk" \
+  >  --alignment "refine/aligned.fasta" \
+  >  --metadata "$TMP/metadata.tsv" \
+  >  --output-tree "$TMP/tree.nwk" \
+  >  --output-node-data "$TMP/branch_lengths.json" \
+  >  --timetree \
+  >  --year-bounds 2000 2020 \
+  >  --coalescent opt \
+  >  --date-confidence \
+  >  --date-inference marginal \
+  >  --clock-filter-iqd 4 \
+  >  --seed 314159 \
+  >  --divergence-units mutations &> /dev/null
+
+Check that the inferred date is 2020-12-31.
+TODO: replace grep with jq? It would require an extra dev dependency.
+
+  $ grep -F "PAN/CDC_259359_V1_V3/2015" -A 3 "$TMP/branch_lengths.json"
+      "PAN/CDC_259359_V1_V3/2015": {
+  .* (re)
+  .* (re)
+        "date": "2020-12-31",
```

Comment:
Yeah, I don't like the precariousness of `-A 3` covering the gap between the sequence name and its `"date"` field.

jq might be nice to have generally, but to avoid crossing that bridge now, how about using a Python one-liner? something like:

```
python3 -c 'import json, sys; print(json.load(sys.stdin)["nodes"]["PAN/CDC_259359_V1_V3/2015"]["date"])' < "$TMP/branch_lengths.json"
```

Reply:
Done: 8f6cbbf32504ce63d13f6ad935808e722e047129